resources/assets/components/CampaignSingle/index.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { flatMap, keyBy, map, sample, forEach, reject, filter } from 'lodash';
+
+import InboxItem from '../InboxItem';
+import ModalContainer from '../ModalContainer';
+import HistoryModal from '../HistoryModal';
+import PostFilter from '../PostFilter';
+import StatusCounter from '../StatusCounter';
+import { RestApiClient} from '@dosomething/gateway';
+
+class CampaignSingle extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const posts = keyBy(flatMap(props.signups, signup => {
+      return signup.posts.map(post => {
+        post.signup = signup;
+        return post;
+      });
+    }), 'id');
+
+    this.state = {
+      posts: posts,
+      filteredPosts: posts,
+      postTotals: props.post_totals,
+      displayHistoryModal: false,
+      historyModalId: null,
+    };
+
+    this.filterPosts = this.filterPosts.bind(this);
+
+    this.api = new RestApiClient;
+    this.updateQuantity = this.updateQuantity.bind(this);
+    this.showHistory = this.showHistory.bind(this);
+    this.hideHistory = this.hideHistory.bind(this);
+  }
+
+  filterPosts(status) {
+    const posts = filter(this.state.posts, {'status' : status.toLowerCase()});
+
+    this.setState({ filteredPosts: posts });
+  }
+
+  // Open the history modal of the given post
+  // @TODO - Move this into the InboxItem component so
```

Comment:
I'm not sure InboxItem would be the best place for this, since the data that's being edited from the modal makes sense at this level for filtering and displaying it in the list, and individual inbox items would have a hard time tracking if a modal was already open.